### DISCUSSION TITLE
Solving certificate problem behind firewall/VPN

### DISCUSSION THREAD

**Discussion by antoniopaolini** [vim-denops/denops.vim#416] [Q&A]
I tried to install denops plugin (to use [bufpreview.vim](https://github.com/kat0h/bufpreview.vim)), but it didn't works.
I received error ` Server stopped 4 times ...d to avoid infinity restart loop.`

After a little bit of tinkering I started vim verbose with a log file, with the command  `vim -V9myVim.log` and I got the complete error:
```
Messages maintainer: The Vim Project
[denops] error: JSR package manifest for '@lambdalisue/workerio' failed to load. Import 'https://jsr.io/@lambdalisue/workerio/meta.json' failed: error sending request for url (https://jsr.io/@lambdalisue/workerio/meta.json): client error (Connect): invalid peer certificate: UnknownIssuer
[denops]     at file:///C:/Users/antopaol/vimfiles/bundle/denops.vim-main/denops/@denops-private/cli.ts:4:8
[denops] Server stopped (1). Restarting...
....[three more times]....
[denops] Server stopped 4 times ...d to avoid infinity restart loop.
```

I thought the problem should be the security software we use at work (if I check TLS certificate in the browser for jsr website I see all signed by Zscaler).
I found [this website](https://medium.com/deno-the-complete-reference/3-ways-of-accepting-self-signed-certificates-in-deno-9c9286ab957) then I tried the last way (I'm in windows 11). **IT WORKS!!!**

Then I added in my vimrc the following line:
```
let $DENO_TLS_CA_STORE='system'
```
Then it works like a charm.

***
I did it in the best way? Any suggestion?

***
I write here because I didn't found anything on the web that helped me; I hope this will be helpful for everyone with my same troubles.

**Comment by lambdalisue** [comment]
Thanks for sharing your Tips 🎉 I didn't know `DENO_TLS_CA_STORE` environment variable.

It seems that variable has introduced from Deno 1.13
https://deno.com/blog/v1.13#use-system-certificate-store-for-tls
